src/network/networkd-dhcp4.c

```diff
@@ -865,7 +865,8 @@ static int dhcp4_update_address(Link *link, bool announce) {
         addr->cinfo.ifa_prefered = lifetime;
         addr->cinfo.ifa_valid = lifetime;
         addr->prefixlen = prefixlen;
-        addr->broadcast.s_addr = address.s_addr | ~netmask.s_addr;
+        if (prefixlen <= 30)
+                addr->broadcast.s_addr = address.s_addr | ~netmask.s_addr;
```

Comment:
i think this really could use a comment explaining why 30 is the threshold

Reply:
See https://tools.ietf.org/html/rfc3021.